src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClass.java

```diff
@@ -127,3 +129,52 @@ private SomeClass() {}
         private SomeClass() {}
     }
 }
+
+class test8 {
+    public static final test8 ONE = new test8() {
+        @Override
+        public int value() {
+            return 1;
+        }
+    };
+
+    private test8() {
+    }
+
+    public int value() {
+        return 0;
+    }
+}
+
+class test {
+
+    private test(String s) {
+        String a = "hello" + s;
+    }
+
+    public int count() {
+        return 1;
+    }
+    public static final test ONE = new test("world");
+    public static final test6 TWO = new test6() {
+    };
+
+    public static void main(String[] args) {
+
+        ArrayList<String> foo = new ArrayList<>();
+        foo.add("world");
+        foo.forEach(test::new);
+
+    }
+}
+
+interface Test1 {
+    ArrayList<String> foo = new ArrayList<>();
+}
+
+// abstract classes cannot be final, see bug #837012
+abstract class test9 {
+    private test9() {
+    }
+}
```

Comment:
Please add `// ok` and `// violation` comments to lines where should be a violation and where it is ok

Reply:
done.